src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -1254,23 +1254,22 @@ FESurfaceDistanceEvaluator::computeSignedDistanceSurfaceMesh(int n_idx, int d_id
                     avg_proj += proj;
 
                     // Get the nodes
-                    const libMesh::Point& n0 = elem->point(0);
-                    const libMesh::Point& n1 = elem->point(1);
-                    X_node[0][0] = n0(0);
-                    X_node[0][1] = n0(1);
-                    X_node[1][0] = n1(0);
-                    X_node[1][1] = n1(1);
-
-#if (NDIM == 3)
-                    const libMesh::Point& n2 = elem->point(2);
-                    X_node[2][0] = n2(0);
-                    X_node[2][1] = n2(1);
-                    X_node[2][2] = n2(2);
+                    const unsigned int n_nodes = elem->n_nodes();
+#if !defined(NDEBUG)
+                    TBOX_ASSERT(n_nodes == NDIM);
 #endif
+                    for (unsigned int i = 0; i < n_nodes; ++i)
+                    {
+                        for (int d = 0; d < NDIM; ++d)
+                        {
+                            X_node[i][d] = (elem->point(i))(d);
+                        }
+                    }
 
```

Comment:
@drwells : You would like this 😛

Reply:
This is much nicer!

Reply:
There is similar code circa line 1190 that should also be generalized to an arbitrary number of nodes.

Reply:
That one is intentional.